internal/blocksync/reactor.go

```diff
@@ -313,6 +326,40 @@ func (r *Reactor) processBlockSyncCh(ctx context.Context, blockSyncCh *p2p.Chann
 	}
 }
 
+// autoRestartIfBehind will check if the node is behind the max peer height by
+// a certain threshold. If it is, the node will attempt to restart itself
+func (r *Reactor) autoRestartIfBehind(ctx context.Context) {
+	if r.blocksBehindThreshold == 0 {
+		r.logger.Info("blocks behind threshold is 0, not checking if node is behind")
+		return
+	}
+
+	for {
+		select {
+		case <-time.After(r.blocksBehindCheckInterval):
+			selfHeight := r.store.Height()
+			maxPeerHeight := r.pool.MaxPeerHeight()
+			threshold := int64(r.blocksBehindThreshold)
+			behindHeight := maxPeerHeight - selfHeight
+			// No peer info yet so maxPeerHeight will be 0
+
+			blockSyncIsSet := r.blockSync.IsSet()
+			if maxPeerHeight == 0 || behindHeight < threshold || blockSyncIsSet {
+				r.logger.Debug("blocks behind does not exceed threshold", "threshold", threshold, "behindHeight", behindHeight, "maxPeerHeight", maxPeerHeight, "selfHeight", selfHeight, "blockSyncIsSet", blockSyncIsSet)
```

Comment:
can we update this log to say "does not exceed threshold or is already in block sync mode"